force_bdss/core/verifier.py

```diff
@@ -116,31 +146,74 @@ def _check_data_source(data_source_model, layer_number):
         if info.name == '':
+            row_index_errors.append(idx)
+    if row_index_errors != []:
+        err_no_string = multi_error_format(row_index_errors)
+        if len(row_index_errors) == 1:
+            errors.append(VerifierError(
+                subject=data_source_model,
+                local_error="Undefined name for input parameter "
+                            "{}".format(err_no_string),
+                global_error="An input parameter is undefined in {} "
+                             "(Layer {})".format(factory.name, layer_number)))
+        else:
             errors.append(VerifierError(
                 subject=data_source_model,
-                error="Undefined name for input "
-                      "parameter {} from {} in layer {}".format(idx,
-                                                                factory.name,
-                                                                layer_number)))
+                local_error="Undefined name for input parameters "
+                            "{}".format(err_no_string),
+                global_error="An input parameter is undefined in {} "
+                             "(Layer {})".format(factory.name, layer_number)))
 
     if len(output_slots) != len(data_source_model.output_slot_info):
         errors.append(VerifierError(
             subject=data_source_model,
-            error="Missing output slot name assignment "
-                  "in layer {}".format(layer_number)))
+            global_error="Missing output slot name assignment "
+                         "in layer {}".format(layer_number)))
 
+    row_index_errors = []
     for idx, info in enumerate(data_source_model.output_slot_info):
         if info.name == '':
-            errors.append(VerifierError(
+            row_index_errors.append(idx)
+    err_no_string = multi_error_format(row_index_errors)
+    if len(row_index_errors) == 1:
+        errors.append(VerifierError(
                 subject=data_source_model,
-                error="Undefined name for output "
-                      "parameter {} from {} in layer {}".format(idx,
-                                                                factory.name,
-                                                                layer_number)))
+                local_error="Undefined name for output "
+                            "parameters {}".format(err_no_string),
+                global_error="An output parameter is undefined in {}"
+                             " (Layer {})".format(factory.name, layer_number)))
 
     return errors
+
+
+def multi_error_format(index_list):
+    """Takes a list of integers and returns a string where they are grouped
+    consecutively wherever possible.
+    For example an input of [0,1,2,4,5,7] returns the string '0-2, 4-5, 7' """
+    index_list.sort()
+    # Single, consecutive or non-consecutive
+    if len(index_list) == 1:
+        return str(index_list[0])
+    else:
```

Comment:
you don't need the else clause here. you are returning in the first case. Keeps the code less indented.